### PR TITLE
Accept xlsx files as animal and sample tables

### LINKED ISSUE #165: Load animals and sample via Excel file
### Feature Inspiration

The [templates for animal and sample data](https://docs.google.com/spreadsheets/d/1To3495KxJkAtnAD9KVdzc162zKbNiYuPEVyPseQPjMQ/edit?usp=sharing) are now a Google Sheet document and researchers find it much simpler to submit an Excel file for this information rather than multiple `.csv`. files. Being able to import from Excel files will make this process much simpler.

### Feature Description

Enhance the `load_animals_and_samples` command to accept _either_ a single Excel spreadsheet with sheets for Animals and Samples, or two `.csv` files, one for each of animals and samples.

### Alternatives Considered

None

### Comment

None

-----

### Assumptions

- Sheet names will be static for now ("Animals" and "Samples")

### Requirements

- XLSX files will be accepted with the same column types and data checks are before
- Other tests should run unaltered

### Limitations

- This will not perform additional data integrity checks
- This will not include any additional features or enhancements around Treatments, Studies, or Tissues (any enhancements in those areas will be separate).

### Affected/Changed Components

- load_animals_and_samples.py

#### GUI Change description

None

#### Code Change Description *(Pseudocode optional)*

Read the tables in either from two Excel sheets or two csv files and pass to import code the same way

#### Tests

Test using a copy of the small_obob sample and animal data reformatted in the combined Excel format (using the template as a guide).

### PR DESCRIPTION
**Pull request by lparsons**
## Summary Change Description

The template animal and sample tables are in a spreadsheet and
researchers find it simpler to provide spreadsheet files instead of csv
files for the animal and sample data, thus accepting those directly is
much simpler.

The default table headers were updated to reflect the standardized
naming conventions and match those in the animal an sample template
files provided to the researchers.

## Affected Issue Numbers

- Resolves #165 

## Code Review Notes

* The `DefaultSampleTableHeaders` now match those used in the [template file](https://docs.google.com/spreadsheets/d/1To3495KxJkAtnAD9KVdzc162zKbNiYuPEVyPseQPjMQ/edit?usp=sharing) which also match those used in the [TraceBase Terms and Definitions](https://docs.google.com/spreadsheets/d/1iOHcBc2bY5vmdFfzam9-C3XkLsnoBt2WCZOG1qmzy4A/edit?usp=sharing)
* The output from the custom command (some of it, not the parts in `utils.py`) now uses `self.stdout.write` instead of `print` as suggested in the [Django Custom Commands documentation](https://docs.djangoproject.com/en/3.2/howto/custom-management-commands/#management-commands-output).
* Errors throw `CommandError` excptions as specified in the [docs](https://docs.djangoproject.com/en/3.2/howto/custom-management-commands/#command-exceptions).

## Checklist

- [x] All issue requirements satisfied (or no linked issues)
- [x] [Linting passes](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting).
- [x] [Migrations created & committed *(or no model changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)
- [x] [Tests implemented *(or no code changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
- [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)

### CODE REVIEW

File: DataRepo/management/commands/load_animals_and_samples.py [hunk 1]
```diff
@@ -41,33 +49,83 @@ def add_arguments(self, parser):
...
+            animal_table_filename,
             dtype={headers.ANIMAL_NAME: str, headers.ANIMAL_TREATMENT: str},
-            keep_default_na=False,
+            sheet="Animals",
+        )
+
+        self.stdout.write(self.style.MIGRATE_HEADING("Loading samples..."))
+        samples = read_from_file(
+            filename=sample_table_filename,
+            dtype={headers.ANIMAL_NAME: str},
```
Comment (jcmatese) [inline DataRepo/management/commands/load_animals_and_samples.py:85]: with the removal of `headers.ANIMAL_TREATMENT: str` , you _may_ encounter issues if the user provides a null value (and pandas converts it to `NaN` or whatever missing value, which then _might_ be created as a protocol name string).  I am not positive about that, but there was a reason why I added it, at the time...

perhaps `keep_default_na=False` makes it work alright...

File: DataRepo/management/commands/load_animals_and_samples.py [hunk 2]
```diff
@@ -41,33 +49,83 @@ def add_arguments(self, parser):
...
+            sheet="Samples",
         )
 
         # merge the two files/dataframes together on Animal ID
+        self.stdout.write(
+            self.style.MIGRATE_HEADING("Merging animal and samples tables...")
```
Comment (jcmatese) [inline DataRepo/management/commands/load_animals_and_samples.py:91]: Q: First time I have seen `self.style.MIGRATE_HEADING`.  I did not find docs, at first google?

**Review by jcmatese — APPROVED** [review]
Seems like it should work, and add some user flexibility.  Only a few questions...

File: DataRepo/management/commands/load_animals_and_samples.py (hunk 1 above)
Comment (lparsons) [reply DataRepo/management/commands/load_animals_and_samples.py:85]: Actually, this is only for the the sample sheet, which does not contain the `ANIMAL_TREATMENT` heading. It is still there when loading the animals sheet (see https://github.com/Princeton-LSI-ResearchComputing/tracebase/pull/166/files#diff-855c443927836173a7da7793332d17a3238e8c3c5b49c19ad7759e4d31f0d32fR78)

File: DataRepo/management/commands/load_animals_and_samples.py (hunk 2 above)
Comment (lparsons) [reply DataRepo/management/commands/load_animals_and_samples.py:91]: Documentation is here: https://docs.djangoproject.com/en/3.2/howto/custom-management-commands/#django.core.management.BaseCommand.style

**Comment by lparsons** [comment]
@fkang-pu Let me know if you want to take a look at this or if I should just merge it. Thanks!

**Review by fkang-pu — APPROVED** [review]
Sorry for the delay. Looks good to me.

**Pull request merged**